packages/@icon-magic/generate/src/plugins/svg-light-dark.ts

```diff
@@ -0,0 +1,179 @@
+              }
+            },
+            {
+              removeViewBox: false
+            }
+          ],
+          js2svg: { pretty: true, indent: 2 }
+        });
+
+        if (lightFlavorContents) {
+          // create optimized light/dark svg assets
+          const lightAsset = await lightSvgo.optimize(lightFlavorContents); // .svg asset's getContents() returns a string
+          const darkAsset = await darkSvgo.optimize(darkFlavorContents);
+
+          const mixedParentSvg = await domSvgParentGenerate(lightAsset, darkAsset);
+
+          const mixedSvgo = new Svgo({
+            plugins: [
+              {
+                removeViewBox: false
+              }
+            ],
+            js2svg: { pretty: true, indent: 2 }
+          });
+
+          const mixedString = serializeToString(mixedParentSvg);
+          const mixedAsset = await mixedSvgo.optimize(mixedString);
+
+          // write the parent svg with light/dark children svg's to the output directory
+          await fs.ensureDir(outputPath);
+          await fs.writeFile(
+            `${path.join(outputPath, imageSet)}-mixed.svg`,
+            mixedAsset.data,
+            {
+              encoding: 'utf8'
+            }
+          );
+
+          // Final new Mixed Flavor
+          const mixedFlavor: Flavor = new Flavor(icon.iconPath, {
+            name: `${imageSet}-mixed`,
+            path: `./${imageSet}-mixed.svg`,
+            colorScheme: 'mixed'
+          });
+
+          // Add new mixed flavor to icon.flavors. Is then added to resulting iconrc.json file.
+          icon.flavors.set(
+            `${imageSet}-mixed`,
+            mixedFlavor
+          );
+          return mixedFlavor;
+        }
+      }
+    }
+    return flavor;
+  }
+};
+
+// TODO: lightAsset type would be `OptimizedSvg` from Svgo interface, but it does not accept this type, I think it is not exported.
+
+/**
+ * Takes two OptimizedSvg's from Svgo interface, adds them as children to a
+ * parent SVG Dom element and returns the Dom element
+ * @param lightAsset `OptimizedSvg` from Svgo interface
+ * @param darkAsset `OptimizedSvg` from Svgo interface
+ * @returns SVGSVGElement Dom elment with the two parameters as children.
+ */
+async function domSvgParentGenerate(lightAsset: any, darkAsset: any): Promise<SVGSVGElement> {
+  // Create SVG PARENT
+  const DOM = new DOMImplementation();
+  const doctype = DOM.createDocumentType(
+    'svg',
+    '-//W3C//DTD SVG 1.1//EN',
+    'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'
+  );
+  const SVG_NS = 'http://www.w3.org/2000/svg';
+
+  // Create an SVG Document and set its doctype
+  const DOCUMENT = DOM.createDocument(SVG_NS, 'svg', doctype);
+
+  // Create SVG element
+  const svgEl = DOCUMENT.createElementNS(SVG_NS, 'svg');
+  svgEl.setAttribute('xmlns', 'http://www.w3.org/2000/svg');
+  svgEl.setAttribute('xmlns:xlink', 'http://www.w3.org/1999/xlink');
+  svgEl.setAttribute('version', '1.1');
```

Comment:
Do we need this for every asset?

Reply:
no actually we only have the first one (155) in our assets, Mario you can remove the rest.